apps/req.c

```diff
@@ -147,14 +148,66 @@ const OPTIONS req_options[] = {
     {NULL}
 };
 
+/*
+ * Is the |kv| key already duplicated?  This is remarkably tricky to get
+ * right.  Return 0 if unique, -1 on runtime error; 1 if found or a syntax
+ * error.
+ */
+static int duplicated(STACK_OF(OPENSSL_STRING) *list, char *kv)
```

Comment:
How about using a LHASH instead of a stack?
The strings are only ever checked for existence and a hash is ideal for that.
Plus, a hash of strings is already defined in libcrypto.

I doubt the performance difference will make a big difference here but avoiding a future gotcha is often good.

Reply:
Right you are.  Additional commit pushed (to be squashed).